java/adapter/avro/src/main/java/org/apache/arrow/consumers/CompositeAvroConsumer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.consumers;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.avro.io.Decoder;
+
+/**
+ * Composite consumer which hold all consumers.
+ * It manages the consume and cleanup process.
+ */
+public class CompositeAvroConsumer implements AutoCloseable {
+
+  private final List<Consumer> consumers;
+
+  public CompositeAvroConsumer(List<Consumer> consumers) {
+    this.consumers = consumers;
+  }
+
+  /**
+   * Consume decoder data and write into {@link VectorSchemaRoot}.
+   */
+  public void consume(Decoder decoder, VectorSchemaRoot root) throws IOException {
+    int valueCount = 0;
+    while (true) {
```

Comment:
lets fix in one of the follow-ups.  but I think looping should be done outside of the consumer interfaces.

Reply:
Thank you Micah, will fix in next PR. I plan to start two PRs one by one(support all type, iterator API), then I think i could be called a initial version and hope to catch up 0.15 branch cut.